pkg/tfbridge/walk.go

```diff
@@ -114,8 +121,98 @@ func propertyPathToSchemaPathInner(
 	return nil
 }
 
+// Translate a a bridged provider schema path into a Pulumi property path.
+//
+// The function hides the complexity of mapping Terraform names to Pulumi property names, joining Schema with user
+// overrides in SchemaInfos, and accounting for MaxItems=1 situations where Pulumi flattens collections to plain values.
+// and therefore SchemaPath values are longer than the PropertyPath values.
+//
+// SchemaPathToPropertyPath may return nil if there is no matching schema found. This may happen when drilling down to
+// values of unknown type, attributes not tracked in schema, or when there is a type mismatch between the path and the
+// schema.
+func SchemaPathToPropertyPath(
```

Comment:
This semantically makes me uneasy just looking at the types, because PropertyPath represents paths in values while SchemaPath represents paths in types. What's the deal with map/list/set elements?   Skimming the implementation it looks like SchemaPaths pointing to list/map/set elements will translate to property paths with '*', for example:

```
			NewSchemaPath().GetAttr("x").Element().GetAttr("y") => x.*.y
```

This could make sense but perhaps call this `SchemaPathToPropertyPathPattern` and/or emphasize in the doc string?

Reply:
Property paths support `*` as a glob element. I'll update the doc comment to call this out explicitly.

Reply:
Well yeah they probably should not, a separate type would be better, but some of the code that accepts PropertyPath actually accepts patterns. Sounds good.